### ISSUE TITLE
[Bug Report][3.3.8] v-virtual-scroll broken

### ISSUE THREAD

**Issue by srguiwiz2** [vuetifyjs/vuetify#17801]
### Environment
**Vuetify Version:** 3.3.8
**Last working version:** 3.3.6
**Vue Version:** 3.3.4
**Browsers:** Google Chrome
**OS:** Ubuntu undefined

### Steps to reproduce
Use v-virtual-scroll with any length simple items, but more so with more, e.g. 150, 250.

### Expected Behavior
All items show when the user scrolls.

### Actual Behavior
When you scroll to top or bottom, some items are not shown.

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNqlUd1uwiAUfhXClSYCW9ySpdFFn8N6gfaYMYESoF1M03ffgdpF643Jrlo+vp9zPnYd3TrH2wZoQVcRjNMywmdpCVm1TDqXf/PhWNsolQV/hTLYKh8bqVk4+lprUii0COuS5m9J/6hIHs1Jy4KuY1HBSTY6IrkjiU565BPbmAN40o1YT1bibqzsJabJ45h4cT9nQoY17nzwiELlIgkQm2FLFIaYYwNZkx3nfOu9vMxe31/m/AyXMJvvk80gRAldUGVc7SMz0vHvUFsssUtWWEC+wAoKkpGEYcvpXNKvGF0ohDhWFmUVaNV6biEK64zYIE34xkZlgFW12Sz5kr+JSoV4C3MIhh18/RPAo0lJFzcxUZ0uT0RdmTnhY0i4QkzLQ0gRD9YC81rwzIOtwIN/dqOJ7HarydXDZim9L22PfWtlz2FS9THkmnfjkP9aOZslJ3zqnvYLmufE5Cyg+1/aMBJx)

### Other comments
It is totally reproducible for us.  It just takes scrolling up and down, maybe a bit faster, maybe just anyway, and it happens.  In my playground example I made it show numbers 0 to 149, and in one trial it stopped at 131, in the other at 136.  When going back to Vuetify 3.3.6 it works as expected.

**Comment by yuwu9145** [comment]
It's probably not a bug, `item-height` is 48px by default, but it is 24px in this reproduction link as well as most demos in our doc.

After specifying `:item-height="24"`, it calculates correctly

https://play.vuetifyjs.com/#eNqlUUFuwjAQ/IrlE0jYbguVqggqeAfhEMhSXGzHsp1UKMrfu3ZIBeGC1FOyszszu+NtSzfW8qYGmtFlAG1VEeAzN4QsG1ZYm35TcahMKKQBd4US2EgX6kIxf3CVUiSTKOFXOU3fnPYAO4H8OgWE3xY5/aOjwGBIGuZVFbISjkWt4mRLIpN0OE9MrffgSDtgHVmKu1WTlhhvM6yOjfvdI9KfdqeDJRKlDcRDqPvLkehDsvVkRbac841zxWXy+v4y5We4+Ml0F2V6IlLojEptKxeYLiz/9pXBYNsohaGkBsaSkYREDJOPdU5PIVifCXEoDdJKULJx3EAQxmqxxjHhahOkBlZWej3nc74QpfThFubgNdu76seDQ5Gczm5sgjxenrC6TiaHj97hCjFV7H20eJAW6NeAYw5MCQ7csxeNaLdXjVoPl0X3Ljcd5q2kOftR1AefYt4OS/7r5CQWlfCpO9rNaNoTnROB7n4BXxgYvw==

**Comment by KaelWD** [comment]
It's actually not responding to page scroll at all, it only adds the listener with `renderless`: https://github.com/vuetifyjs/vuetify/blob/eb235561b867fed0b94b2273843f2e1cf7a957c6/packages/vuetify/src/components/VVirtualScroll/VVirtualScroll.tsx#L73

**Comment by yuwu9145** [comment]
scroll doesn't trigger is due to scroll parent height is the full content height as opposed to window height in this case.

There seems to be two issues around scroll container height and first/last index calculation when in big height. I am diving into it...

**Comment by srguiwiz2** [comment]
I tried it again with Vuetify 3.3.20 and today even the first demo at https://vuetifyjs.com/en/components/virtual-scroller/ loses elements at the beginning and at the end when scrolling back and forth somewhat.

I looked at the source and stepped through some of `packages/vuetify/src/composables/virtual.ts` and `packages/vuetify/src/components/VVirtualScroll/VVirtualScroll.tsx` but there were more conditions than I was able to keep track of.

**Comment by KaelWD** [comment]
@srguiwiz2 try the build from #18392

**Comment by srguiwiz2** [comment]
That was good for the first time looking at a list of items, loaded after starting our app.  It scrolled well, up and down, all items.

Then I tried to look at different content listed, loading a different list of items, and then it only could show the first item.  Our app wasn't defective, and this is how I verified that:  I switched to 3.3.20 and it was able to load other lists as it should, but of course the scrolling in 3.3.20 was missing some items.

Getting closer!  Thank you for the effort.

TL;DR

Btw, the npm of the fix caused dependency problems, so I actually did a build of branch `fix/virtual-scroll-dynamic-window`, commit `d3b4d29e1`...

**Comment by KaelWD** [comment]
Can you reproduce this in https://play.vuetifyjs.com/?

> the npm of the fix caused dependency problems

Did you install it as an alias? https://vuetifyjs.com/en/getting-started/installation/#nightly-builds

**Comment by srguiwiz2** [comment]
I just made this test https://play.vuetifyjs.com/#eNp9UcFOwzAM/RUrEqLVaNZuDMG0ISFOnDkyDqFzISJNoyQtQlX/HSfZkAaCS5z4PT+/2E8jc7ae3xnDhx7Zmm08tkYJj7c7DbAZCmFMvMZH3WkvpEZ7SMXkIK3vhSpcbTulYC1Jwm13LMYd+6YS+SgOQ+FU59d7bESvPJFHCHSYiJ9u4zEzwWZ+YioqzX/2PZok4NRlyKRPnOjQkwql8eDQ9+mPsjWd9TCCxQYmaGzXwjnN5TyApOp89OQq2MIT5/zOWvGZVasy5+/46bL8mbfCZAjbW3j0VurXDPP8R/HitPjvWh76378Je9/tMbtawQx+q5EYmc2SrQgq9ETotUdLYBlSTa9rLzsN7oNa5DCmWckGsgNzNoMzWHwjhAVBPgjVk6OD8YRNgMrhv8zqwAwhHjTih9CGWFnwcAFVWZZkdzNPS4gLcah9GrqAtnuRCmGPg6xxp9l0wZZ8yS9ZitVNYWxRXS9vFnxR16t9c8WevwCimuVp but I think you have fixed it since the last post.
